geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/attribute/AttributeQueryableIndex.scala

```diff
@@ -235,12 +236,23 @@ trait AttributeQueryableIndex extends AccumuloFeatureIndex with LazyLogging {
       attributeScan.columnFamilies, recordThreads, hasDupes, joinFunction, joinQuery)
   }
 
-  override def getFilterStrategy(sft: SimpleFeatureType, filter: Filter): Seq[AccumuloFilterStrategyType] = {
+  override def getFilterStrategy(sft: SimpleFeatureType,
+                                 filter: Filter,
+                                 transform: Option[SimpleFeatureType]): Seq[AccumuloFilterStrategyType] = {
+    import AttributeIndex.AllowJoinPlans
     val attributes = FilterHelper.propertyNames(filter, sft)
-    val indexedAttributes = attributes.filter(a => Option(sft.getDescriptor(a)).exists(_.isIndexed))
-    indexedAttributes.flatMap { attribute =>
+    val indexedAttributes = attributes.flatMap { a =>
+      Option(sft.getDescriptor(a)).map(_.getIndexCoverage()).toSeq.collect {
+        case coverage if coverage != IndexCoverage.NONE => (a, coverage)
+      }
+    }
+    indexedAttributes.flatMap { case (attribute, coverage) =>
       val (primary, secondary) = FilterExtractingVisitor(filter, attribute, sft, attributeCheck)
-      if (primary.isDefined) {
+      lazy val joinCheck = coverage match {
+        case IndexCoverage.FULL => true
+        case IndexCoverage.JOIN => AllowJoinPlans.get || !requiresJoin(sft, attribute, secondary, transform)
+      }
+      if (primary.isDefined && joinCheck) {
```

Comment:
I'll add some clarifying comments, but yeah I think the logic needs to stay split for the reason you say.